document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -82,6 +83,38 @@ public boolean upsert(Key key, Document document) throws IOException {
     }
   }
 
+  @Override
+  public boolean upsert(Key key, Document document, Filter condition, @Nullable Boolean isUpsert)
+      throws IOException {
+    StringBuilder upsertQueryBuilder =
+        new StringBuilder(getUpsertSQL(isUpsert != null ? isUpsert : true));
+
+    String jsonString = prepareUpsertDocument(key, document);
+    Params.Builder paramsBuilder = Params.newBuilder();
+    paramsBuilder.addObjectParam(key.toString());
+    paramsBuilder.addObjectParam(jsonString);
+    paramsBuilder.addObjectParam(jsonString);
```

Comment:
accidentally, added the `jsonString` twice? Or, is it deliberate?

Reply:
It is deliberate and required.